supervision/detection/core.py

```diff
@@ -805,6 +806,52 @@ def from_paddledet(cls, paddledet_result) -> Detections:
             class_id=paddledet_result["bbox"][:, 0].astype(int),
         )
 
+    @classmethod
+    def from_lmm(cls, lmm: Union[LMM, str], result: str, **kwargs) -> Detections:
+        """
+        Creates a Detections object from the given result string based on the specified
+        Large Multimodal Model (LMM).
+
+        Args:
+            lmm (Union[LMM, str]): The type of LMM (Large Multimodal Model) to use.
+            result (str): The result string containing the detection data.
+            **kwargs: Additional keyword arguments required by the specified LMM.
+
+        Returns:
+            Detections: A new Detections object.
+
+        Raises:
+            ValueError: If the LMM is invalid, required arguments are missing, or
+                disallowed arguments are provided.
+            ValueError: If the specified LMM is not supported.
+
+        Examples:
+            ```python
+            import supervision as sv
+
+            paligemma_result = "<loc0256><loc0256><loc0768><loc0768> cat"
```

Comment:
I think this would be also useful for models like KOSMOS-2 so best to make it very general (this is a trend with VLMs these days) https://huggingface.co/docs/transformers/en/model_doc/kosmos-2#transformers.Kosmos2ForConditionalGeneration.forward.example

Reply:
So what I want to do is one `from_lmm` function, providing separate dedicated parsers for each model. :)